artifacts/artifacts.go

```diff
@@ -0,0 +1,47 @@
+package artifacts
+
+import (
+	"errors"
+	"fmt"
+	"io"
+
+	"github.com/flanksource/artifacts"
+	dutyAPI "github.com/flanksource/duty/api"
+	"github.com/flanksource/duty/context"
+	"github.com/flanksource/duty/models"
+	"github.com/flanksource/incident-commander/api"
+	"github.com/google/uuid"
+	"gorm.io/gorm"
+)
+
+// UploadArtifact Uploads the given artifact data to the default artifact store
+func UploadArtifact(ctx context.Context, artifactID string, reader io.ReadCloser) error {
```

Comment:
I am a little confused by this function - Don't we upload to the artifact store and then insert the db record ?

Reply:
It's the other way around - the artifact db records are first pushed and then the data